src-vue/src/utils/ui.ts

```diff
@@ -0,0 +1,26 @@
+import { ElNotification, NotificationHandle } from "element-plus";
+import { i18n } from "../main";
+
+/**
+ * Displays content to the user in the form of a notification appearing on screen bottom right.
+ **/
+function showNotification(
+    title: string,
+    message: string = '',
+    type: 'success' | 'warning' | 'error' | 'info' = 'success',
+    duration: number = 4500
+): NotificationHandle {
+    return ElNotification({
+        title, message, type, duration,
+        position: 'bottom-right',
+    });
+}
+
+/**
+ * Helper method displaying an error message to the user.
+ **/
+function showErrorNotification(error: string): NotificationHandle {
+    return showNotification(i18n.global.tc('generic.error'), error, 'error');
+}
```

Comment:
Does TypeScript have optional function arguments like e.g. Python?

Cause then we could give `title` a default value of `i18n.global.tc('generic.error')` and if an error message has a special title, then we set it explicitly just for that.

Reply:
We could do that, but we should put `title` argument after the `error` one; if you want to see a working example, check `showNotification` prototype:
```typescript
function showNotification(
    title: string,
    message: string = '',
    type: 'success' | 'warning' | 'error' | 'info' = 'success',
    duration: number = 4500
)
```

Reply:
Would be fine by me ^^

Reply:
Addressed in f41241b3a612f6807d229f7f125505bece495eda